zap/src/output/typescript/mod.rs

```diff
@@ -3,6 +3,8 @@ use crate::config::{Config, Enum, Ty};
 pub mod client;
 pub mod server;
 
+const TUPLE_BECOMES_ARRAY_THRESHOLD: i8 = 11;
```

Comment:
Let's turn this into a configuration option.

Reply:
What should it be called? `typescript_tuples_become_arrays_threshold`, or is that too verbose? Luau might offer tuple types someday and I don't want people to be confused.

Reply:
I think we'll need to document it either way (see `docs/options.md`).
Maybe `typescript_array_threshold`?